umodbus/asynchronous/common.py

```diff
@@ -0,0 +1,244 @@
+#!/usr/bin/env python
+#
+# Copyright (c) 2019, Pycom Limited.
+#
+# This software is licensed under the GNU GPL version 3 or any
+# later version, with permitted additional terms. For more information
+# see the Pycom Licence v1.0 document supplied with this file, or
+# available at https://www.pycom.io/opensource/licensing
+#
+
+# system packages
+from ..sys_imports import List, Optional, Tuple, Union
+
+# custom packages
+from .. import functions, const as Const
+from ..common import CommonModbusFunctions, Request
+
+
+class AsyncRequest(Request):
+    """Asynchronously deconstruct request data received via TCP or Serial"""
+
+    async def send_response(self,
+                            values: Optional[list] = None,
+                            signed: bool = True) -> None:
+        """
+        Send a response via the configured interface.
+
+        :param      values:  The values
+        :type       values:  Optional[list]
+        :param      signed:  Indicates if signed values are used
+        :type       signed:  bool
+        """
+        await self._itf.send_response(self,
```

Comment:
not sure whether this really works and requires the `self` parameter. I also recommend to use named parameters